plugins/processors/gpuattributes/internal/awsneuron_metric_checker.go

```diff
@@ -0,0 +1,46 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package internal
+
+import (
+	"strings"
+)
+
+const (
+	CONTAINER_NEURON_CORE_METRIC_PREFIX   = "container_neuroncore_"
+	NODE_NEURON_CORE_METRIC_PREFIX        = "node_neuroncore_"
+	POD_NEURON_CORE_METRIC_PREFIX         = "pod_neuroncore_"
+	CONTAINER_NEURON_DEVICE_METRIC_PREFIX = "container_neurondevice_"
+	NODE_NEURON_DEVICE_METRIC_PREFIX      = "node_neurondevice_"
+	POD_NEURON_DEVICE_METRIC_PREFIX       = "pod_neurondevice_"
+	NODE_NEURON_METRIC_PREFIX             = "node_neuron_"
+)
+
+type AwsNeuronMetricChecker struct {
+}
+
+func NewAwsNeuronMetricChecker() *AwsNeuronMetricChecker {
+	return &AwsNeuronMetricChecker{}
+}
+
+func (md *AwsNeuronMetricChecker) IsProcessedNeuronMetric(name string) bool {
+	switch {
```

Comment:
Are there any edge cases in neuron metric names that is preventing us to use a simple regex match?

Reply:
converting it to a regex, since the cases were finite I thought a simple switch would've been fine, will convert it to a regex an see if it looks better